File: old/mailtidy/agent/executor.py
"""Execution boundary for applying planned tool calls."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class ToolObservation:
    """Result returned from a tool call and merged back into AgentState."""

    tool_name: str
    ok: bool
    summary: str
    payload: dict[str, object] | None = None

File: src/agent/executor.ts
/**
 * Tool 执行边界：把 plan 落地到具体 tool call，并把 observation merge 回 state（Phase 1 占位）。
 */

export interface ToolObservation {
  toolName: string;
  ok: boolean;
  summary: string;
  payload?: Record<string, unknown>;
}
